
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [1]:
%matplotlib inline

import pandas as pd
import matplotlib
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

### Question 1 - What test is appropriate for this problem? Does CLT apply?

#### Answer
    Considering both the t-test and z-test, one would think that the t-test is more appropriate for this problem.
    Ideally we would use the z-test, but only 1 of the two conditions are met.
    1. Yes, sample size(n) > 30 
    2. No, population standard deviation is not known
    Therefore we use the t-test.
    
    CLT applies in this context because the sample size is large. 

### Question 2 - What are the null and alternate hypotheses?

#### Answer
    Null(H0)- There is no discrimination in the labour market of the U.S.A based on whether a name is black-sounding or white-sounding. Ie. The response rates are the same with factors considered.
    Alternate(H1) - There is discrimination in the labour market of the U.S.A based on whether a name is black-sounding or white-sounding. Ie. The callback rates are the different. 
    

### Question 3 - Compute margin of error, confidence interval, and p-value.

In [35]:
dfb = data[data['race']=='b']
dfw = data[data['race']=='w']
callback = data['call']

In [37]:
moe = 1.96 * np.std(callback)/np.sqrt(len(callback))
print "Margin of error is", moe

Margin of error is 0.00764086832945


In [38]:
avg = callback.mean()
print (avg-moe,avg+moe)


(0.072851948535994943, 0.088133685194893424)


In [34]:
t_stat, pval = stats.ttest_ind(dfw['call'], dfb['call'])
t_stat, pval

(4.1147052908617514, 3.9408021031288859e-05)

#### Answer - Question 3

Margin of error is 0.76%

Confidence Interval is (7.2%, 8.8%)

P-value is 3.94e-05



### Question 4 - Discuss statistical significance.

In [57]:
print "Callback rates for people with black-sounding names - ", dfb['call'].mean()
print "Callback rates for people with black-sounding names - ", dfw['call'].mean()

Callback rates for people with black-sounding names -  0.0644763857126
Callback rates for people with black-sounding names -  0.0965092405677


In [42]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [55]:
titles = list(data.columns.values)


In [46]:
for i in titles:
    try:
        t_stat2, pval2 = stats.ttest_ind(dfw[i], dfb[i])
        print (i,t_stat2, pval2)
        
    except:
        pass

('education', 0.24047517576654476, 0.80997203176721733)
('ofjobs', 0.17629292771974545, 0.86007115112888888)
('yearsexp', 0.18461970685747395, 0.85353501824812827)
('honors', 0.44856358183187639, 0.65376647488431394)
('volunteer', -0.40759024305759978, 0.68359245176859607)
('military', -1.1128831243101491, 0.26581357972093489)
('empholes', 0.28809583635970071, 0.7732856444597449)
('occupspecific', -0.52157539040120759, 0.60198972963193875)
('occupbroad', -0.21794295813518974, 0.82748269062094948)
('workinschool', -0.20201250014774666, 0.83991539316968733)
('email', -0.057356200828648833, 0.95426380623027018)
('computerskills', -2.1664271042751966, 0.030326933955391936)
('specialskills', 0.21348739187623014, 0.83095576251836034)
('h', 0.0, 1.0)
('l', 0.0, 1.0)
('call', 4.1147052908617514, 3.9408021031288859e-05)
('adid', 0.0, 1.0)
('fracblack', nan, nan)
('fracwhite', nan, nan)
('lmedhhinc', nan, nan)
('fracdropout', nan, nan)
('fraccolp', nan, nan)
('linc', nan, nan)
('col', -0.5102719

In [50]:
for i in titles:
    try:
        t_stat2, pval2 = stats.ttest_ind(dfw[i], dfb[i])
        if pval2 < 0.05:
            print (i,t_stat2, pval2)
        
    except:
        pass


('computerskills', -2.1664271042751966, 0.030326933955391936)
('call', 4.1147052908617514, 3.9408021031288859e-05)


In [54]:
print "black names -", dfb['computerskills'].mean()
print "white names -", dfw['computerskills'].mean()

black names - 0.832443531828
white names - 0.808624229979


### Answer - Question 4
From question 3 we have:
Margin of error is 0.76%, 
Confidence Interval is (7.2%, 8.8%), 
pvalue is 3.94e-05

#### P-value suggests that we cannot accept the Null hypothesis. 

The callback rate for white-sounding names is 9.65%, where as for black-sounding names, the rate is 6.45%.
Looking to our confidence interval, that would entail that callback rates for white names are abnormally high, and call back rates for black names are abnormally low. 

##### Maybe people with black names are less qualified?
Conducting students T-test on all necessary columns(variables), only one qualification had statistical significance.
Computer skills had a p-value less than 0.05, 0.03 to be concise.

##### Is it possible that people with black sounding names have less/worse computer skill and that accounts for the abnormally low callback rates?
Interestingly, a significantly higher percentage of people with black-sounding names have computer skills compared to that of people with white-sounding names, 83.2% and 80.1% respectively. 

#### This leads us to conclude and accept the alternate hypothesis. 
There is discrimination in the labour market of the U.S.A based on whether a name is black-sounding or white-sounding. Ie. The callback rates are the different.


